packages/app/cypress/e2e/integration/runs.spec.ts

```diff
@@ -1,80 +1,256 @@
+      cy.contains('projectId: \'abcdef\'')
+    })
+
+    it('displays correct source control instructions', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Ensure that your cypress.config.js file is checked into source control:')
+      cy.get('[data-cy="copy-button"]').first().click()
+      cy.contains('Copied!')
+    })
+
+    it('displays correct instructions on how to record a run', () => {
+      cy.loginUser()
+      cy.remoteGraphQLIntercept(async (obj) => {
+        if (obj.result.data?.cloudProjectsBySlugs) {
+          for (const proj of obj.result.data.cloudProjectsBySlugs) {
+            if (proj.runs?.nodes) {
+              proj.runs.nodes = []
+            }
+          }
+        }
+
+        return obj.result
+      })
+
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.contains('Run this command in your local development terminal or in CI:')
+      cy.contains('--record --key 2aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa')
+      cy.get('[data-cy="copy-button"]').last().click()
+      cy.contains('Copied!')
+    })
   })
 
-  it('if logged in and connected', { viewportWidth: 1200 }, () => {
-    cy.loginUser()
-    cy.visitApp()
-    cy.intercept('mutation-ExternalLink_OpenExternal', { 'data': { 'openExternal': true } }).as('OpenExternal')
+  context('Runs - Runs List', () => {
+    it('displays a list of recorded runs if a run has been recorded', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
+      cy.get('[data-cy="runs"]')
+    })
+
+    it('displays each run with correct information', () => {
+      cy.loginUser()
+      cy.visitApp()
+      cy.get('[href="#/runs"]').click()
 
-    cy.get('[href="#/runs"]').click()
-    cy.contains('a', 'OVERLIMIT').click()
-    cy.wait('@OpenExternal')
-    .its('request.body.variables.url')
-    .should('equal', 'http://dummy.cypress.io/runs/4')
+      cy.get('[href="http://dummy.cypress.io/runs/0"]').first().within(() => {
+        cy.findByText('fix: make gql work CANCELLED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
+
+      cy.get('[href="http://dummy.cypress.io/runs/1"]').first().within(() => {
+        cy.findByText('fix: make gql work ERRORED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
+
+      cy.get('[href="http://dummy.cypress.io/runs/2"]').first().within(() => {
+        cy.findByText('fix: make gql work FAILED')
+        cy.get('[data-cy="run-card-icon"]')
+      })
+
+      cy.get('[href="http://dummy.cypress.io/runs/0"]').first().get('[data-cy="run-card-author"]').contains('John Appleseed')
```

Comment:
This selector with first should really be avoided...
This being said, you should at least alias it with `cy.get('[href="http://dummy.cypress.io/runs/0"]').first().as('firstRun')`

then you can call it `cy.get('@firstRun')`. It should simplify your code a little.